sdk-lib/src/main/java/cash/z/ecc/android/sdk/internal/db/pending/PendingTransactionDb.kt

```diff
@@ -24,11 +27,67 @@ import kotlinx.coroutines.flow.Flow
     entities = [
         PendingTransactionEntity::class
     ],
-    version = 1,
+    version = 2,
     exportSchema = true
 )
 abstract class PendingTransactionDb : RoomDatabase() {
     abstract fun pendingTransactionDao(): PendingTransactionDao
+
+    companion object {
+
+        /*
+         * Non-automatic migration required because to_address became nullable.
+         */
+        internal val MIGRATION_1_2 = object : Migration(1, 2) {
+            override fun migrate(database: SupportSQLiteDatabase) {
+                database.execSQL(
+                    """
+                 ALTER TABLE pending_transactions RENAME TO pending_transactions_old;
+                 CREATE TABLE pending_transactions(
+                     id                         INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL,
+                     toAddress                  TEXT,
+                     toInternalAccountIndex     INTEGER,
+                     accountIndex               INTEGER NOT NULL,
+                     minedMeight                INTEGER,
+                     expiryHeight               INTEGER,
+                     cancelled                  INTEGER,
+                     encodeAttempts             INTEGER DEFAULT (0),
+                     error_message              TEXT,
+                     errorCode                  INTEGER,
+                     submitAttempts             INTEGER DEFAULT (0),
+                     createTime                 INTEGER,
+                     txid                       BLOB,
+                     value                      INTEGER NOT NULL,
+                     raw                        BLOB,
+                     memo                       BLOB,
+                     fee                        INTEGER NOT NULL
+                 );
+                 INSERT INTO pending_transactions
+                 SELECT
+                     id,
+                     toAddress,
+                     NULL,
+                     accountIndex,
+                     minedHeight,
+                     expiryHeight,
+                     cancelled,
+                     encodeAttempts,
+                     errorMessage,
+                     errorCode,
+                     submitAttempts,
+                     createTime,
+                     txid,
+                     value,
+                     raw,
+                     memo,
+                     ${ZcashSdk.MINERS_FEE}
```

Comment:
Is there the potential for this value to be incorrect as a consequence of a user importing a seed from another wallet? Also, what about wallets that were active prior to the 10000 ZAT -> 1000 ZAT fee change?

Reply:
I was hoping to avoid fee being a nullable column if possible.

How would this be impacted by import from another wallet?

Do we have a block height at which the 10000 ZAT -> 1000 ZAT fee change occurred?

Reply:
The change in zcashd was made in https://github.com/zcash/zcash/releases/tag/v4.2.0, but we don't have any idea when various wallets adopted it, so I don't think it's safe to assume any given height; after all, some wallets are *still* paying the 10000 ZAT fee even today, as we saw when researching fee behavior for ZIP 317.

I think that we should make the column nullable, and then in the future we can attempt to fill that data in properly by actually downloading transactions for which we don't have the fee information, getting their inputs, and actually figuring it out properly.

And  maybe in a future network upgrade we can change the transaction format to include the fee paid explicitly. That would simplify a bunch of things.